### PR TITLE
[PAYLOAD]feat: return all `etablissements` when `all_etablissements` admin option is given

### LINKED ISSUE #316: Ajouter la pagination pour les établissements

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #316 

This PR introduces new functionalities to enhance Elasticsearch document retrieval based on specific requirements. The primary additions include the introduction of an `admin` option with the parameter `include_admin=all_etablissements` and the inclusion of a `page_etablissements` option. The goal is to facilitate the retrieval of Elasticsearch documents by their `id`, particularly in cases where a Siren has more than 100 associated `etablissements`.

In scenarios where a Siren has over 100 `etablissements`, these are indexed in Elasticsearch across multiple documents, each containing 100 `etablissements`. The document ids are structured with an increment of 100, such as siren-100, siren-200, and so on. The `page_etablissements` parameter plays a crucial role in computing the appropriate document id using the formula: `siren_query - page_etablissements * 100`. This approach enables the retrieval of a specific document (representing 100 `etablissements`) with each query.

By implementing these options, the website gains the capability to efficiently fetch and display the relevant `etablissements` associated with a specific document id, addressing the challenge posed by Siren numbers with a large number of associated establishments.

### CODE REVIEW

File: aio/aio-proxy/aio_proxy/tests/e2e_tests/test_api.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def test_fetch_company(api_response_tester):
     test if searching for `la poste` returns the right siren as the first search result.
     """
     path = "search?q=la poste"
-    api_response_tester.test_field_value(path, "siren", "356000000")
+    api_response_tester.test_field_value(path, 0, "siren", "356000000")
     api_response_tester.test_number_of_results(path, min_total_results)
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/tests/e2e_tests/test_api.py:22]: Should not 0 be the default `page_etablissements` value ? Maybe I dont undersatn these `0` you add every where

**Review by XavierJp — CHANGES_REQUESTED** [review]
For me 99% correct ! I think we just miss the metadata : nombre_etablissement etc.

In terms of logic I would maybe see it this way : 
- if is siren -> run the query ans get default object. Lets call it `object1`
- if should_get_doc_by_id -> lets call it `object X`
 - substitute the etablissements of `object 1` with those of `object X`

**Comment by HAEKADI** [comment]
> For me 99% correct ! I think we just miss the metadata : nombre_etablissement etc.
> 
> In terms of logic I would maybe see it this way :
> 
> * if is siren -> run the query and get default object. Lets call it `object1`
> * if should_get_doc_by_id -> lets call it `object X`
> * substitute the etablissements of `object 1` with those of `object X`

I'm not sure I understand. Why would you run two queries one to get object_1 and another to get object_X, when you can just run one and get object_X ?

**Review by XavierJp — APPROVED** [review]

**Comment by XavierJp** [comment]
My bad, this is great. Still dont get the `0`everywhere though

**Comment by HAEKADI** [comment]
@XavierJp 
> My bad, this is great. Still dont get the `0`everywhere though

I've incorporated the ability to specify the desired result in the response for testing purposes. In my initial setup(return multiple docs with same siren in one page instead of just one doc), I aimed to assess multiple results on the same page to ensure consistency with a specific Siren.
I retained this feature in the logic, as I believe it's valuable to maintain the flexibility of selecting which result to test in the response.
If it's causing too much noise in this PR, I could implement it in another one.

**Pull request merged**